### ISSUE TITLE
AssumeRoleWithWebIdentityCredentialsOptions properties incorrectly claim that properties have setters

### ISSUE THREAD

**Issue by zingmars** [aws/aws-sdk-net#2085]
### Describe the issue

The documentation for AssumeRoleWithWebIdentityCredentialsOptions claims that all of the variables have both getter and a setter, however the actual implementation (https://github.com/aws/aws-sdk-net/blob/master/sdk/src/Core/Amazon.Runtime/Credentials/AssumeRoleWithWebIdentityCredentialsOptions.cs) only has getters.

On that note, is there a reason this class only has getters? I don't see any other way to specify the token duration when using AssumeRoleWithWebIdentityCredentials.

### Links

https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/Runtime/TAssumeRoleWithWebIdentityCredentialsOptions.html

**Comment by ashishdhingra** [comment]
@zingmars Thanks for reporting the issue. We would get this corrected.

**Comment by ashishdhingra** [comment]
Hi @zingmars,

Good morning.

Could you please share your use case where you would be using `AssumeRoleWithWebIdentityCredentialsOptions` directly (via `AssumeRoleWithWebIdentityCredentials`), rather than going through the credentials chain via profile resolution?

Thanks,
Ashish

**Comment by zingmars** [comment]
I was just trying to understand how to set the expiry time for the token manually, since I needed to pass the token to another application on EKR and IAM roles were behaving really weirdly. I eventually figured out the issue with IAM roles, so at the end of the day I guess I wouldn't really be using this manually. The documentation saying that the options had setters really threw me off a bit at the time.
